approved/v93k/flow_control.tf

```diff
@@ -1725,38 +1795,24 @@ test_flow
   run_and_branch(pgm_BEA7F3B)
   then
   {
-    @T5_BEA7F3B_RAN = 1;
+    run(read0_BEA7F3B);
   }
   else
   {
     stop_bin "", "fail", , bad, noreprobe, red, 10, over_on;
     @T5_BEA7F3B_RAN = 1;
   }
-  if @T5_BEA7F3B_RAN == 1 then
-  {
-    run(read0_BEA7F3B);
-  }
-  else
-  {
-  }
   print_dl("Test the block form of if_ran");
   run_and_branch(pgm_1_BEA7F3B)
   then
   {
-    @T6_BEA7F3B_RAN = 1;
-  }
-  else
-  {
-    stop_bin "", "fail", , bad, noreprobe, red, 10, over_on;
-    @T6_BEA7F3B_RAN = 1;
-  }
-  if @T6_BEA7F3B_RAN == 1 then
-  {
     run(read0_1_BEA7F3B);
     run(read0_2_BEA7F3B);
   }
   else
   {
+    stop_bin "", "fail", , bad, noreprobe, red, 10, over_on;
+    @T6_BEA7F3B_RAN = 1;
   }
```

Comment:
Patched the FlagOptimizer processor to ignore the RAN flags.  Created a new processor to remove the set-result node in the fail branch for if_ran conditionals.